test/gas-analytics/HubPool.RefundExecution.ts

```diff
@@ -0,0 +1,136 @@
+
+    // Seed data worker with bond tokens.
+    await seedWallet(dataWorker, [], weth, consts.bondAmount.mul(10));
+    await weth.connect(dataWorker).approve(hubPool.address, consts.bondAmount.mul(10));
+
+    // Deploy test tokens for each chain ID
+    for (let i = 0; i < REFUND_CHAIN_COUNT; i++) {
+      destinationChainIds.push(i);
+      l1Tokens[i] = [];
+      for (let j = 0; j < REFUND_TOKEN_COUNT; j++) {
+        const _l1Token = await deployErc20(owner, `Test Token #${i}`, `T-${i}`);
+        l1Tokens[i].push(_l1Token);
+
+        // Mint data worker amount of tokens needed to bond a new root
+        await seedWallet(dataWorker, [_l1Token], undefined, consts.bondAmount);
+        await _l1Token.connect(dataWorker).approve(hubPool.address, consts.bondAmount);
+
+        // Mint LP amount of tokens needed to cover relay
+        await seedWallet(liquidityProvider, [_l1Token], undefined, STARTING_LP_AMOUNT);
+        await enableTokensForLP(owner, hubPool, weth, [_l1Token]);
+        await _l1Token.connect(liquidityProvider).approve(hubPool.address, STARTING_LP_AMOUNT);
+        await hubPool.connect(liquidityProvider).addLiquidity(_l1Token.address, STARTING_LP_AMOUNT);
+      }
+    }
+  });
+
+  it(`Tree with 1 Leaf: ${REFUND_SEND_COUNT} total transfers, ${REFUND_TOKEN_COUNT} different tokens`, async function () {
+    const { leafs, tree } = await constructSimpleTree();
+    const leafIndexToExecute = 0;
+
+    await hubPool.connect(dataWorker).initiateRelayerRefund(
+      [3117], // bundleEvaluationBlockNumbers used by bots to construct bundles. Length must equal the number of leafs.
+      1, // poolRebalanceLeafCount. There is exactly one leaf in the bundle.
+      tree.getHexRoot(), // poolRebalanceRoot. Generated from the merkle tree constructed before.
+      consts.mockDestinationDistributionRoot, // destinationDistributionRoot. Not relevant for this test.
+      consts.mockSlowRelayFulfillmentRoot // Mock root because this isn't relevant for this test.
+    );
+
+    // Advance time so the request can be executed and execute the request.
+    await timer.setCurrentTime(Number(await timer.getCurrentTime()) + consts.refundProposalLiveness);
+    const txn = await hubPool
+      .connect(dataWorker)
+      .executeRelayerRefund(leafs[leafIndexToExecute], tree.getHexProof(leafs[leafIndexToExecute]));
+
+    // Balances should have updated as expected for tokens contained in the first leaf.
+    for (let i = 0; i < REFUND_TOKEN_COUNT; i++) {
+      expect(await l1Tokens[leafIndexToExecute][i].balanceOf(hubPool.address)).to.equal(
+        STARTING_LP_AMOUNT.sub(SEND_AMOUNT)
+      );
+      expect(await l1Tokens[leafIndexToExecute][i].balanceOf(mockAdapter.address)).to.equal(SEND_AMOUNT);
+    }
+
+    // Now that we've verified that the transaction succeeded, let's compute average gas costs.
+    const receipt = await txn.wait();
+    console.log(receipt.cumulativeGasUsed);
```

Comment:
perhaps add some additional strings to this? something like `cumlativeGasUsed:` or something.

Reply:
Yeah just starting from here, I'll log it